ste/mgr-JobMgr.go

```diff
@@ -704,6 +708,12 @@ func (jm *jobMgr) DeferredCleanupJobMgr() {
 	jm.logger.CloseLog()
 }
 
+func (jm *jobMgr) ProcessMsg() {
+	jm.jstm.processMsg <- struct{}{}
+	close(jm.jstm.xferDone)
+	<-jm.jstm.doneProcessMsg
```

Comment:
There is a risk that if handleStatusUpdateMessage() go routine has exited, then this will be stuck indefinitely.
We may not have that situation in our case due to the other things we take care, but let's try to avoid that even for other users who may GetJobProgress() at any time.

What we can do is the following:
In handleStatusUpdateMessage(), before exiting close the doneProcessMsg channel, so here if handleStatusUpdateMessage() has exited the read from that doneProcessMsg won't block.